src/rules/completedDocsRule.ts

```diff
@@ -395,11 +395,31 @@ class CompletedDocsWalker extends Lint.ProgramAwareRuleWalker {
     }
 
     public visitVariableDeclaration(node: ts.VariableDeclaration): void {
-        this.checkNode(node, ARGUMENT_VARIABLES);
+        this.checkVariable(node);
         super.visitVariableDeclaration(node);
     }
 
-    private checkNode(node: ts.NamedDeclaration, nodeType: DocType, requirementNode: ts.Declaration = node): void {
+    private checkVariable(node: ts.VariableDeclaration) {
+        // Only check variables in variable declaration lists
+        // and not variables in catch clauses and for loops.
+        if (node.parent === undefined || node.parent.kind !== ts.SyntaxKind.VariableDeclarationList) {
```

Comment:
`node.parent` as well as `list.parent` and `statement.parent` will never be undefined. You can simply assert that using for example `node.parent!.kind`